### PR TITLE
bazel: update to gazel v13 and create sources rules throughout tree

### PR DESCRIPTION
**Pull request by ixdy**
**What this PR does / why we need it**: creates filegroup rules using https://github.com/mikedanese/gazel/pull/20, which we can then use in creating release tarballs

**Special notes for your reviewer**: this obviously should not be merged before https://github.com/mikedanese/gazel/pull/20 is merged and tagged; this is more to give you an idea of what the output looks like. The verification tests should fail this PR in any case.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by spxtr** [comment]
BUILD.bazel
```
filegroup(
    name = "all-srcs",
    srcs = [
        ":package-srcs",
        "//api:all-srcs",
        "//build:all-srcs",
        "//cluster:all-srcs",
        "//cmd:all-srcs",
        "//docs:all-srcs",
        "//examples:all-srcs",
        "//federation:all-srcs",
        "//hack:all-srcs",
        "//pkg:all-srcs",
        "//plugin:all-srcs",
        "//test:all-srcs",
        "//third_party:all-srcs",
        "//vendor:all-srcs",
    ],
    tags = ["automanaged"],
)
```
pkg/BUILD
```
filegroup(
    name = "package-srcs",
    srcs = glob(["**"]),
    tags = ["automanaged"],
    visibility = ["//visibility:private"],
)

filegroup(
    name = "all-srcs",
    srcs = [
        ":package-srcs",
        "//pkg/admission:all-srcs",
        "//pkg/api:all-srcs",
        "//pkg/apimachinery:all-srcs",
        "//pkg/apis/abac:all-srcs",
        "//pkg/apis/apps:all-srcs",
        "//pkg/apis/authentication:all-srcs",
        "//pkg/apis/authorization:all-srcs",
        "//pkg/apis/autoscaling:all-srcs",
        "//pkg/apis/batch:all-srcs",
        "//pkg/apis/certificates:all-srcs",
        "//pkg/apis/componentconfig:all-srcs",
        "//pkg/apis/extensions:all-srcs",
        "//pkg/apis/imagepolicy:all-srcs",
        "//pkg/apis/meta/v1:all-srcs",
        "//pkg/apis/policy:all-srcs",
        "//pkg/apis/rbac:all-srcs",
        "//pkg/apis/storage:all-srcs",
        "//pkg/auth/authenticator:all-srcs",
        "//pkg/auth/authorizer:all-srcs",
        "//pkg/auth/group:all-srcs",
        "//pkg/auth/handlers:all-srcs",
        "//pkg/auth/user:all-srcs",
        "//pkg/capabilities:all-srcs",
        "//pkg/client/cache:all-srcs",
        "//pkg/client/chaosclient:all-srcs",
        "//pkg/client/clientset_generated/clientset:all-srcs",
        "//pkg/client/clientset_generated/internalclientset:all-srcs",
        "//pkg/client/conditions:all-srcs",
        "//pkg/client/informers/informers_generated:all-srcs",
        "//pkg/client/leaderelection:all-srcs",
        "//pkg/client/listers/apps/internalversion:all-srcs",
        "//pkg/client/listers/apps/v1beta1:all-srcs",
        "//pkg/client/listers/authentication/internalversion:all-srcs",
        "//pkg/client/listers/authentication/v1beta1:all-srcs",
        "//pkg/client/listers/authorization/internalversion:all-srcs",
        "//pkg/client/listers/authorization/v1beta1:all-srcs",
        "//pkg/client/listers/autoscaling/internalversion:all-srcs",
        "//pkg/client/listers/autoscaling/v1:all-srcs",
        "//pkg/client/listers/batch/internalversion:all-srcs",
        "//pkg/client/listers/batch/v1:all-srcs",
        "//pkg/client/listers/batch/v2alpha1:all-srcs",
        "//pkg/client/listers/certificates/internalversion:all-srcs",
        "//pkg/client/listers/certificates/v1alpha1:all-srcs",
        "//pkg/client/listers/core/internalversion:all-srcs",
        "//pkg/client/listers/core/v1:all-srcs",
        "//pkg/client/listers/extensions/internalversion:all-srcs",
        "//pkg/client/listers/extensions/v1beta1:all-srcs",
        "//pkg/client/listers/imagepolicy/internalversion:all-srcs",
        "//pkg/client/listers/imagepolicy/v1alpha1:all-srcs",
        "//pkg/client/listers/policy/internalversion:all-srcs",
        "//pkg/client/listers/policy/v1alpha1:all-srcs",
        "//pkg/client/listers/policy/v1beta1:all-srcs",
        "//pkg/client/listers/rbac/internalversion:all-srcs",
        "//pkg/client/listers/rbac/v1alpha1:all-srcs",
        "//pkg/client/listers/storage/internalversion:all-srcs",
        "//pkg/client/listers/storage/v1beta1:all-srcs",
        "//pkg/client/metrics:all-srcs",
        "//pkg/client/record:all-srcs",
        "//pkg/client/restclient:all-srcs",
        "//pkg/client/retry:all-srcs",
        "//pkg/client/testdata:all-srcs",
        "//pkg/client/testing/cache:all-srcs",
        "//pkg/client/testing/core:all-srcs",
        "//pkg/client/transport:all-srcs",
        "//pkg/client/typed/discovery:all-srcs",
        "//pkg/client/typed/dynamic:all-srcs",
        "//pkg/client/unversioned:all-srcs",
        "//pkg/cloudprovider:all-srcs",
        "//pkg/controller:all-srcs",
        "//pkg/conversion:all-srcs",
        "//pkg/credentialprovider:all-srcs",
        "//pkg/dns:all-srcs",
        "//pkg/fieldpath:all-srcs",
        "//pkg/fields:all-srcs",
        "//pkg/generated:all-srcs",
        "//pkg/genericapiserver:all-srcs",
        "//pkg/healthz:all-srcs",
        "//pkg/httplog:all-srcs",
        "//pkg/hyperkube:all-srcs",
        "//pkg/kubeapiserver:all-srcs",
        "//pkg/kubectl:all-srcs",
        "//pkg/kubelet:all-srcs",
        "//pkg/kubemark:all-srcs",
        "//pkg/labels:all-srcs",
        "//pkg/master:all-srcs",
        "//pkg/metrics:all-srcs",
        "//pkg/probe:all-srcs",
        "//pkg/proxy:all-srcs",
        "//pkg/quota:all-srcs",
        "//pkg/registry:all-srcs",
        "//pkg/routes:all-srcs",
        "//pkg/runtime:all-srcs",
        "//pkg/security:all-srcs",
        "//pkg/securitycontext:all-srcs",
        "//pkg/selection:all-srcs",
        "//pkg/serviceaccount:all-srcs",
        "//pkg/ssh:all-srcs",
        "//pkg/storage:all-srcs",
        "//pkg/types:all-srcs",
        "//pkg/util:all-srcs",
        "//pkg/version:all-srcs",
        "//pkg/volume:all-srcs",
        "//pkg/watch:all-srcs",
    ],
    tags = ["automanaged"],
)
```

**Comment by ixdy** [comment]
The `pkg_tar` rule is kinda terrible (it's slow and chmod's everything `0555` by default), but it's a neat POC:
```console
{0} [1513] (gazel-srcs) ~/.../src/k8s.io/kubernetes $ make clean
+++ [0103 15:15:38] Verifying Prerequisites....
+++ [0103 15:15:39] Removing _output directory
{0} [1515] (gazel-srcs) ~/.../src/k8s.io/kubernetes $ bazel clean
INFO: Starting clean (this may take a while). Consider using --expunge_async if the clean takes more than several minutes.
{0} [1515] (gazel-srcs) ~/.../src/k8s.io/kubernetes $ bazel build kubernetes-src.tar.gz
INFO: Found 1 target...
Target //:kubernetes-src.tar.gz up-to-date:
  bazel-bin/kubernetes-src.tar.gz
INFO: Elapsed time: 44.928s, Critical Path: 38.49s
{0} [1516] (gazel-srcs) ~/.../src/k8s.io/kubernetes $ bazel build kubernetes-src.tar.gz
INFO: Found 1 target...
Target //:kubernetes-src.tar.gz up-to-date:
  bazel-bin/kubernetes-src.tar.gz
INFO: Elapsed time: 1.623s, Critical Path: 0.61s
{130} [1516] (gazel-srcs) ~/.../src/k8s.io/kubernetes $ tar tzvf bazel-bin/kubernetes-src.tar.gz | wc -l
16934
{0} [1517] (gazel-srcs) ~/.../src/k8s.io/kubernetes $ echo "something new" >> README.md
{0} [1517] (gazel-srcs) ~/.../src/k8s.io/kubernetes $ bazel build kubernetes-src.tar.gz
INFO: Found 1 target...
Target //:kubernetes-src.tar.gz up-to-date:
  bazel-bin/kubernetes-src.tar.gz
INFO: Elapsed time: 36.420s, Critical Path: 35.99s
```

File: hack/update-bazel.sh [hunk 1]
```diff
@@ -20,5 +20,5 @@ set -o pipefail
 export KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
 source "${KUBE_ROOT}/hack/lib/init.sh"
 
-go get -u gopkg.in/mikedanese/gazel.v8/gazel
+go get -u gopkg.in/mikedanese/gazel.v11/gazel
```
Comment (ixdy) [inline hack/update-bazel.sh]: vendoring gazel is turning out to be quite a pain.

**Comment by mikedanese** [comment]
We might want to get https://github.com/mikedanese/gazel/pull/25 in first. Is this tested on Mac yet?

**Comment by ixdy** [comment]
@mikedanese the [`IsDir()` check](https://github.com/mikedanese/gazel/blob/master/gazel/gazel.go#L581) should handle this correctly for Kubernetes. AFAIK we don't have any `build` files (yet).

**Comment by ixdy** [comment]
this PR conflicted with #39444, looks like, which is to be expected.

**Comment by ixdy** [comment]
@mikedanese tested this on Mac, seems to work fine.

**Comment by ixdy** [comment]
Rebased.

**Comment by ixdy** [comment]
now using gazel v12.

**Comment by ixdy** [comment]
I don't understand why that failed so quickly...

@k8s-bot bazel test this

**Comment by ixdy** [comment]
gazel v13: now with less /tmp pollution!

**Comment by ixdy** [comment]
@k8s-bot test this

**Pull request merged**